pubtools/_pulp/tasks/push/items/erratum.py

```diff
@@ -18,24 +18,50 @@
 class ErratumUploadContext(UploadContext):
     """A custom context for Erratum uploads.
 
-    This context object avoids having to query the UPLOAD_REPO repeatedly.
+    This context object avoids having to query the PushItem's repo repeatedly.
     """
 
     upload_repo = attr.ib(default=None)
 
 
+class ErratumPushItemException(BaseException):
+    """
+    Custom exception for PulpErratumPushItem specific issues.
+
+    Mainly used for when a year value can't be parsed from the advisory name.
+    """
+
+    pass
+
+
 @supports_type(ErratumPushItem)
 @attr.s(frozen=True, slots=True)
 class PulpErratumPushItem(PulpPushItem):
     """Handler for errata."""
 
-    # Erratums are always uploaded to this repo first.
-    # Pulp stores the erratum pkglist for (erratum_id, repo) pair. If an erratum is
-    # uploaded for an already existing ID, the pkglist is overwritten if the repo is
-    # same, else is appended. To keep the pkglist consistent, all the erratums are
-    # uploaded to the same repo instead of a random repo from the list of destinations.
-    # For more details, refer RHELDST-12782.
-    UPLOAD_REPO = "all-rpm-content"
+    MULTI_UPLOAD_CONTEXT = True
+    ADVISORY_PATTERN = re.compile(r"RH.A-(\d{4})", re.IGNORECASE)
+    CONTENT_SPLIT_RANGES = [(2014, 2030), (8014, 8030)]
```

Comment:
The other side of the code is being updated to use 2040 as the end of the range.